app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomAdapter.kt

```diff
@@ -74,12 +77,20 @@ class ChatRoomAdapter(private val context: Context,
             imageUnknownAvatar.setVisibility(true)
         }
 
-        private fun bindUserName(message: Message, textView: TextView) = message.sender?.username.let {
-            textView.textContent = it.toString()
-        }.ifNull {
-            textView.textContent = context.getString(R.string.msg_unknown)
-        }
+        private fun bindUserName(message: Message, textView: TextView) {
```

Comment:
For now we can live with this logic inside the Adapter, but in the future, when we have local persistence, I'd like to use a mapper to deliver a  "MessageViewModel" to the adapter that is really ready to be "rendered", All the information should be ready, including spannables with links, emojis, etc...

Reply:
Agreed.